### PR TITLE
[SPARK-44903][PYTHON][DOCS] Refine docstring of `approx_count_distinct`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr refine docstring of `approx_count_distinct ` and add some new examples.

### Why are the changes needed?
To improve PySpark documentation

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass Github Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @allisonwang-db @zhengruifeng FYI

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -3669,38 +3669,83 @@ def approxCountDistinct(col: "ColumnOrName", rsd: Optional[float] = None) -> Col
...
 
     Returns
     -------
     :class:`~pyspark.sql.Column`
-        the column of computed results.
+        A new Column object representing the approximate unique count.
+
+    References
+    ----------
+    .. seealso:: :func:`count_distinct`
```
Comment (zhengruifeng) [inline python/pyspark/sql/functions.py:3698]: it seems the pyspark docstrings normally use the `See Also` section

File: python/pyspark/sql/functions.py [hunk 2]
```diff
@@ -3669,38 +3669,83 @@ def approxCountDistinct(col: "ColumnOrName", rsd: Optional[float] = None) -> Col
...
+
+    See Also
+    ----------
+    :meth:`pyspark.sql.functions.count_distinct`
 
     Examples
     --------
+    Example 1: Counting distinct values in a single column DataFrame representing integers
+
+    >>> from pyspark.sql.functions import approx_count_distinct
```
Comment (zhengruifeng) [inline python/pyspark/sql/functions.py:3704]: what about using this style of import in the new examples?
```
>>> from pyspark.sql import functions as F
>>> df = spark.createDataFrame([1,2,2,3], "INT")
>>> df.agg(F.approx_count_distinct("value").alias('distinct_values')).show()
```

I want to make all examples and docstrings follow this way.

Comment (LuciferYang) [reply python/pyspark/sql/functions.py]: done

File: python/pyspark/sql/functions.py [hunk 3]
```diff
@@ -3669,38 +3669,83 @@ def approxCountDistinct(col: "ColumnOrName", rsd: Optional[float] = None) -> Col
...
+    :meth:`pyspark.sql.functions.count_distinct`
 
     Examples
     --------
+    Example 1: Counting distinct values in a single column DataFrame representing integers
+
+    >>> from pyspark.sql import functions as F
     >>> df = spark.createDataFrame([1,2,2,3], "INT")
-    >>> df.agg(approx_count_distinct("value").alias('distinct_values')).show()
+    >>> df.agg(F.approx_count_distinct("value").alias('distinct_values')).show()
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions.py:3706]: Actually I think we don't need to add `from pyspark.sql import functions as F` here. We can just assume that users are reading `pyspark.sql.functions` doc, and assume that they are already imported.

Comment (HyukjinKwon) [reply python/pyspark/sql/functions.py:3706]: cc @allisonwang-db and @zhengruifeng .

File: python/pyspark/sql/functions.py (hunk 2 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/functions.py]: The issue with `from pyspark.sql import functions as F` is that `F` itself actually isn't very Pythonic naming (but we use this for the legacy/historical reason). If we can go back, I would have named it something like `from pyspark.sql import functions as sf`.

Comment (HyukjinKwon) [reply python/pyspark/sql/functions.py]: and second reason is that `from pyspark.sql.functions import approx_count_distinct` is perfectly fine.

Comment (HyukjinKwon) [reply python/pyspark/sql/functions.py]: and .. the last reason is that:
> We can just assume that users are reading pyspark.sql.functions doc, and assume that they are already imported.

File: python/pyspark/sql/functions.py (hunk 3 above)
Comment (LuciferYang) [reply python/pyspark/sql/functions.py:3706]: You're right, but adding `from pyspark.sql import functions as F` makes it easier for users to copy and run test cases.

Comment (HyukjinKwon) [reply python/pyspark/sql/functions.py:3706]: Can we set the small set of rules and policies in the unbrella JIRA at SPARK-44728 before we go further? Would be great if we can have moment, take a look at other docs, and set the template or guides down to the JIRA.

cc @allisonwang-db and @zhengruifeng .

Comment (HyukjinKwon) [reply python/pyspark/sql/functions.py:3706]: At least, [these three reasons](https://github.com/apache/spark/pull/42596#discussion_r1300942444) were my initial intention when I fixed all the docs. If we want sth different from it, we should probably set it down before fixing all the docs.

Comment (LuciferYang) [reply python/pyspark/sql/functions.py:3706]: I’m not professional in Python, but I personally prefer to add `from … import …`.：）

As for which one is better between `from pyspark.sql.functions import approx_count_distinct` and `from pyspark.sql import functions as sfunc`, I can make further changes

File: python/pyspark/sql/functions.py (hunk 2 above)
Comment (zhengruifeng) [reply python/pyspark/sql/functions.py]: @HyukjinKwon sound good. thanks for the explaination.

@LuciferYang sorry, let go back to `from pyspark.sql.functions import approx_count_distinct`

Comment (LuciferYang) [reply python/pyspark/sql/functions.py]: done

Comment (allisonwang-db) [reply python/pyspark/sql/functions.py:3704]: yea I would also prefer `from pyspark.sql.functions import approx_count_distinct`

File: python/pyspark/sql/functions.py [hunk 4]
```diff
@@ -3669,38 +3669,83 @@ def approxCountDistinct(col: "ColumnOrName", rsd: Optional[float] = None) -> Col
...
+    See Also
+    ----------
+    :meth:`pyspark.sql.functions.count_distinct`
 
     Examples
     --------
+    Example 1: Counting distinct values in a single column DataFrame representing integers
+
+    >>> from pyspark.sql.functions import approx_count_distinct
     >>> df = spark.createDataFrame([1,2,2,3], "INT")
```
Comment (allisonwang-db) [inline python/pyspark/sql/functions.py:3705]: Can we use lower case for `INT` here?

File: python/pyspark/sql/functions.py [hunk 5]
```diff
@@ -3669,38 +3669,83 @@ def approxCountDistinct(col: "ColumnOrName", rsd: Optional[float] = None) -> Col
...
+    |distinct_pairs|
+    +--------------+
+    |             3|
+    +--------------+
+
+    Example 4: Counting distinct values with a specified relative standard deviation
+
+    >>> from pyspark.sql.functions import approx_count_distinct
+    >>> df = spark.range(100000)
+    >>> df.agg(approx_count_distinct("id", 0.1).alias('distinct_values')).show()
```
Comment (allisonwang-db) [inline python/pyspark/sql/functions.py:3743]: can we compare the results of approx_count_distinct when 1) using the default rsd and 2) using 0.1 in this example?

File: python/pyspark/sql/functions.py [hunk 6]
```diff
@@ -3669,38 +3669,83 @@ def approxCountDistinct(col: "ColumnOrName", rsd: Optional[float] = None) -> Col
...
+
+    Example 3: Counting distinct values in a DataFrame with multiple columns
+
+    >>> from pyspark.sql.functions import approx_count_distinct, struct
+    >>> df = spark.createDataFrame([("Alice", 1),
+    ...                             ("Alice", 2),
+    ...                             ("Bob", 3),
+    ...                             ("Bob", 3)], ["name", "value"])
+    >>> df = df.withColumn("combined", struct("name", "value"))
+    >>> df.agg(approx_count_distinct(df["combined"]).alias('distinct_pairs')).show()
```
Comment (allisonwang-db) [inline python/pyspark/sql/functions.py:3732]: can we make this example consistent with the others by using `approx_count_distinct("combined")`?

**Review by allisonwang-db — COMMENTED** [review]
Thanks for adding the examples!

File: python/pyspark/sql/functions.py [hunk 7]
```diff
@@ -3669,38 +3669,83 @@ def approxCountDistinct(col: "ColumnOrName", rsd: Optional[float] = None) -> Col
...
     +---------------+
     |distinct_values|
     +---------------+
     |              3|
     +---------------+
+
+    Example 2: Counting distinct values in a single column DataFrame representing strings
+
+    >>> from pyspark.sql.functions import approx_count_distinct
+    >>> df = spark.createDataFrame(["apple", "orange", "apple", "banana"], "string").toDF("fruit")
```
[the remark this answers is not in the document]
Comment (LuciferYang) [reply python/pyspark/sql/functions.py:3716]: like `spark.createDataFrame([("apple"), ("orange"), ("apple"), ("banana")], ['fruit'])`?

File: python/pyspark/sql/functions.py (hunk 4 above)
Comment (LuciferYang) [reply python/pyspark/sql/functions.py:3705]: done

File: python/pyspark/sql/functions.py (hunk 6 above)
Comment (LuciferYang) [reply python/pyspark/sql/functions.py]: done

File: python/pyspark/sql/functions.py (hunk 5 above)
Comment (LuciferYang) [reply python/pyspark/sql/functions.py]: Change to xx

```
>>> df.agg(approx_count_distinct("id").alias('with_default_rsd'),
    ...        approx_count_distinct("id", 0.1).alias('with_rsd_0.1')).show()
    +----------------+------------+
    |with_default_rsd|with_rsd_0.1|
    +----------------+------------+
    |           95546|      102065|
    +----------------+------------+
```
when I am not sure if this is the `compare ` you mentioned.

File: python/pyspark/sql/functions.py [hunk 8]
```diff
@@ -3669,38 +3669,83 @@ def approxCountDistinct(col: "ColumnOrName", rsd: Optional[float] = None) -> Col
...
 @try_remote_functions
 def approx_count_distinct(col: "ColumnOrName", rsd: Optional[float] = None) -> Column:
-    """Aggregate function: returns a new :class:`~pyspark.sql.Column` for approximate distinct count
-    of column `col`.
+    """
+    Applies an aggregate function to return an approximate distinct count of the specified column.
 
-    .. versionadded:: 2.1.0
+    This function returns a new :class:`~pyspark.sql.Column` that estimates the number of distinct
+    elements in a column or a group of columns.
```
[the remark this answers is not in the document]
Comment (LuciferYang) [reply python/pyspark/sql/functions.py]: done

**Review by HyukjinKwon — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]

**Comment by zhengruifeng** [comment]
thanks, merged to master

**Pull request closed without merging**